src/eth1spec/evm/instructions.py

```diff
@@ -62,12 +68,28 @@ def sstore(evm: Evm) -> None:
     OutOfGasError
         If `evm.gas_left` is less than `20000`.
     """
-    evm.gas_left = subtract_gas(evm.gas_left, U256(20000))
-
-    k = pop(evm.stack)
-    v = pop(evm.stack)
-
-    evm.env.state[evm.current].storage[k.to_be_bytes32()] = v
+    key = pop(evm.stack)
+    val = pop(evm.stack)
+    curr_storage = evm.env.state[evm.current].storage
+    key_as_bytes32 = key.to_be_bytes32()
+    existing_val = curr_storage.get(key_as_bytes32, U256(0))
```

Comment:
```suggestion
    existing_val = curr_storage.get(key.to_be_bytes32(), U256(0))
```
would rather just do this where ever we need it

Reply:
We are converting the key to bytes32 in 2 places in the code. Hence just made it as a variable to not compute it twice.

Reply:
I'd rather avoid creating a new variable for this since it's only used twice. If we don't want to `to_be_bytes32()` twice, maybe we append it on the result of `pop(..)`? I find that slightly confusing though, because it's differentiating the data underlying the key and value. I feel like our implementation should avoid these kind of confusions have things look and feel natural so the only thing that should be considered is the spec.

Reply:
As the blockchain progresses, I believe that there would be a lot of EVM calls, and hence I would prefer this optimization though it's minor, given it doesn't affect the readability a lot. I agree with your point that this is used only in 2 places and doesn't need a variable in the POV of readability.

That said, I don't know how to proceed forward regarding this. Let me know your final thoughts and I will modify the code accordingly. Thanks.

Reply:
I think if you update the line `key = pop(evm.stack)` to `key = pop(evm.stack).to_bytes_32()` we'll improve the readability and not lose the performance.

Reply:
Sounds good thanks.